packages/apollo-server-codegen/src/IR/Objects.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  FieldDefinitionNode,
+  FieldNode,
+  ObjectTypeDefinitionNode,
+  ObjectTypeExtensionNode,
+  OperationDefinitionNode,
+  parse,
+  StringValueNode,
+  isTypeExtensionNode,
+  InterfaceTypeDefinitionNode,
+  InterfaceTypeExtensionNode,
+  Kind
+} from "graphql";
+
+import { Translatable, Translator } from "../Translators";
+import { Description } from "./Descriptions";
+import { ResolverDefinition, TypelessResolverDefinition } from "./Resolvers";
+import { CompoundType, makeType, TypeDefinition } from "./Types";
+
+export const SELECTION_OFFSET = 7; // number of chars added in the below parse(`query { ${source} }`) call
+const parseSelections = (source: string) =>
+  (parse(`query { ${source} }`).definitions[0] as OperationDefinitionNode)
```

Comment:
This should probably become a shared utility for parsing a selectionset directly https://github.com/apollographql/apollo-server/blob/master/packages/apollo-gateway/src/utilities/graphql.ts#L100

Reply:
I made it into a shared utility in this package, but not in `apollo-graphql` as per your other comment. Resolving this comment for now.